src/components/Homepage/HomepageStatistics.vue

```diff
@@ -0,0 +1,211 @@
+				key: block.key || index,
+				copy: documentToHtmlString(block.bodyCopy ?? ''),
+				image: this.statsIcons[index],
+			}));
+		},
+		statsIcons() {
+			const icons = this.content?.media?.filter(({ key }) => key.indexOf('homepage-stats-icon') > -1) ?? [];
+			return icons.map(image => ({
+				description: image?.description ?? '',
+				title: image?.title ?? '',
+				url: image?.file?.url ?? ''
+			}));
+		},
+		statsVideo() {
+			// Grabbing the 1x video from contentful. 2x video = 'homepage-statistics-video-2x'
+			// eslint-disable-next-line max-len
+			const video = this.content?.media?.filter(({ key }) => key.indexOf('homepage-statistics-video-1x') > -1) ?? [];
+			return video;
+		},
+		totalLoansInDollarsFormatted() {
+			// Defining empty arrays to put the split contentful strings into
+			const part1 = [];
+			const part2 = [];
+			const contentfulTextString = this.statsBlockText[0].copy;
+			const loansInDollarsFormatted = numeral(this.totalLoansInDollars).format('$0.0a').slice(0, -1);
+			// Splitting the string returned from contentful, where it finds '{value}'
+			const stringSplit = contentfulTextString.split('{value}');
+			// Pushing the two pieces of the split string the respected arrays
+			part1.push(stringSplit[0]);
+			part2.push(stringSplit[1]);
+			// Joining both pieces of the string and the data from graphql
+			const finalString = part1 + loansInDollarsFormatted + part2;
+
+			return finalString;
+		},
+		repaymentRateFormatted() {
+			// Defining empty arrays to put the split contentful strings into
+			const part1 = [];
+			const part2 = [];
+			const contentfulTextString = this.statsBlockText[1].copy;
+			const repaymentRateFormatted = `${numeral(this.repaymentRate).format('0')}%`;
+			// Splitting the string returned from contentful, where it finds '{value}'
+			const stringSplit = contentfulTextString.split('{value}');
+			// Pushing the two pieces of the split string the respected arrays
+			part1.push(stringSplit[0]);
+			part2.push(stringSplit[1]);
+			// Joining both pieces of the string and the data from graphql
+			const finalString = part1 + repaymentRateFormatted + part2;
+
+			return finalString;
+		},
+		numCountriesAndLendersFormatted() {
+			// Defining empty arrays to put the split contentful strings into
+			const part1 = [];
+			const part2 = [];
+			const part3 = [];
+			const contentfulTextString = this.statsBlockText[2].copy;
+			const numberOfLendersFormatted = numeral(this.numLenders).format('0.0b').slice(0, -1);
+			// Splitting the string returned from contentful, where it finds '{value}'
+			const stringSplit = contentfulTextString.split('{value}');
+			// Pushing the three pieces of the split string the respected arrays
+			part1.push(stringSplit[0]);
+			part2.push(stringSplit[1]);
+			part3.push(stringSplit[2]);
+			// Joining all the pieces of the string and the data from graphql
+			const finalString = part1 + this.numCountries + part2 + numberOfLendersFormatted + part3;
```

Comment:
Please work on a utility function to add to contentfulUtils to handle this type of string parsing and concatenation. I think you could automate this with a function that accepts your sourceString, splitKey and an array of injected values.

Something like:
`
buildDynamicString(this.statsBlockText[2].copy, '{value}', [this.numCountries, numberOfLendersFormatted]);

buildDynamicString(sourceString, splitKey dynamicValues) {
  // let newString = '';
  // split the sourceString
  // forEach with index the sourceString
  //   > concat the item to the newString
  //   > if dynamicValue[i] exist concat dynamicValue[i] onto the newString
  }
`

Reply:
+1

Reply:
Thanks for that Matt, I've added this in the latest commit.